client/finality-grandpa/src/authorities.rs

```diff
@@ -631,6 +686,36 @@ impl<H, N: Add<Output=N> + Clone> PendingChange<H, N> {
 	}
 }
 
+// Tracks authority set changes. We store the block numbers for the first block of each authority
+// set.
+#[derive(Debug, Encode, Decode, Clone, PartialEq)]
+pub struct AuthoritySetChanges<N> {
+    authority_set_changes: Vec<(u64, N)>, // (set_id, block_number)
+}
+
+impl<N: Ord + Clone> AuthoritySetChanges<N> {
+    pub(crate) fn empty() -> Self {
+        Self {
+            authority_set_changes: Default::default(),
+        }
+    }
+
+    pub(crate) fn append(&mut self, set_id: u64, block_number: N) {
+        self.authority_set_changes.push((set_id, block_number));
+    }
+
+    pub(crate) fn get_set_id(&self, block_number: N) -> Option<(u64, N)> {
+		let idx = self.authority_set_changes
+			.binary_search_by_key(&block_number, |(_set_id, n)| n.clone())
+			.unwrap_or_else(|b| b);
+		if idx < self.authority_set_changes.len() {
+			Some(self.authority_set_changes[idx].clone())
+		} else {
+			None
+		}
+    }
+}
+
 #[cfg(test)]
```

Comment:
Maybe we can one of the tests that apply standard/forced changes to assert that the authority set change is properly tracked.